test/Npgsql.Tests/Types/RangeTests.cs

```diff
@@ -247,6 +248,23 @@ await using (var conn = await OpenConnectionAsync())
             isNpgsqlDbTypeInferredFromClrType: false);
     }
 
+    [Test]
+    public async Task Ranges_not_supported_with_slim_builder_without_EnableRanges()
+    {
+        await using var slimDataSource = new NpgsqlSlimDataSourceBuilder(ConnectionString).Build();
+        await AssertTypeUnsupportedRead<NpgsqlRange<int>, InvalidCastException>(@"[1,2)", "int4range", slimDataSource);
```

Comment:
#4971 is going to make it so we throw `NotSupportedException` if we attempt to read a record without `EnableRecords`. We probably should align them (return unsupported handler for ranges?)

Reply:
Absolutely, that would be great. This way the user also gets the informative message when they attempt to read a range.

Reply:
Probably in a separate pr. Also, might be a good idea to make a separate resolver with all of the unsupported types (instead of adding them to BuiltInTypeResolver)

Reply:
1. It will make BuiltInTypeResolver a bit faster (for example switch over dataTypes might be a bit more optimized since there will be less values)
2. Just having all of them bundled in a separate file is nice, BuiltInTypeResolver only handles things it supports, UnsupportedTypeResolver only handles things it doesn't support :trollface: